File: CodeWars/Fundamentals/BuildAPileOfCubes/BuildAPileOfCubes.rs

fn find_nb(m: u64) -> i32 {
    let mut n: u64 = 1;

    while (n*(n+1)/2).pow(2) < m {
        n += 1;
    }

    if (n*(n+1)/2).pow(2) == m { n as i32 } else { -1 }
}

fn testing(n: u64, exp: i32) -> () {
    assert_eq!(find_nb(n), exp);
}

fn basics_find_nb() {
    testing(4183059834009, 2022);
    testing(24723578342962, -1);
    testing(135440716410000, 4824);
    testing(40539911473216, 3568);
    testing(26825883955641, 3218);
}

fn main() {
    basics_find_nb();
}

File: CodeWars/Fundamentals/BuildAPileOfCubes/BuildAPileOfCubes.c

#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <assert.h>

long long findNb(long long m) {
    long long n = 1;

    while (pow((n*(n+1)/2.0), 2) < m)
        n++;

    return pow((n*(n+1)/2.0), 2) == m ? n : -1;
}

long long findNb(long long m);

void testequal(long long m, long long  exp)
{
    long long act = findNb(m);
    if(act != exp)
        printf("Error. Expected %lld but got %lld\n", exp, act);
    assert(act == exp);
}

int main() {
    {
        testequal(24723578342962, -1);
        testequal(135440716410000, 4824);
        testequal(26825883955641, 3218);
        testequal(441, 6);
    }
}

